File: An 3/Sem 2/MPP/Java/Server/src/main/java/com/mcostea/SalesAgency/persistance/DAO.java
package com.mcostea.SalesAgency.persistance;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 * Created by mihaicostea on 16/04/15.
 */
public class DAO {
    protected static String dbConnString = "jdbc:mysql://localhost/agency?user=root";
    protected static String dbDriverName = "com.mysql.jdbc.Connection";

    protected static volatile Connection conn;

    public static void connect() {
        try {
            Class.forName(OrdersDAO.dbDriverName);
            OrdersDAO.conn = DriverManager.getConnection(OrdersDAO.dbConnString);
        } catch (ClassNotFoundException ex) {
            Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
        } catch (SQLException ex) {
            Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
        }

    }

    public static void disconnect() {
        try {
            OrdersDAO.conn.close();
        } catch (SQLException ex) {
            Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

}

File: An 3/Sem 2/MPP/CSharp/Server/Server/Persistance/DAO.cs
using System;
using MySql.Data;
using MySql.Data.MySqlClient;
using System.Data;

namespace Server
{
	public class DAO
	{

        protected String dbConnString = "server=localhost;uid=root;pwd=root;database=agency";
		protected volatile MySqlConnection connection;

		public DAO ()
		{
			
		}

		public void Connect() {
			lock (this) {
				try {
					this.connection = new MySql.Data.MySqlClient.MySqlConnection();
					this.connection.ConnectionString = this.dbConnString;
					this.connection.Open();
				} catch (MySqlException ex) {
					Console.WriteLine (ex.ToString ());
				}
			}
		}

		public void Disconnect() {
			lock (this) {
				try {
					this.connection.Close();
				} catch (MySqlException ex) {
					Console.WriteLine (ex.ToString ());
				}
			}
		}
	}
}

